### PR TITLE
Kucoin: Fix unmarshalling of FuturesTicker resp

### PR DESCRIPTION
**Pull request by gbjk**
* Fix several string float64 fields not unmarshalling
* Fix time field misnamed
* Use order.Side for sides
* Add tests
* Rename `FuturesRealTimeTicker` to `FuturesTicker`

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run
- [x] TestGetFuturesRealTimeTicker

### CODE REVIEW

**Review by gloriousCode — APPROVED** [review]
tACK!

**Review by shazbert — APPROVED** [review]
tACK. Nice work!

**Review by thrasher- — CHANGES_REQUESTED** [review]

File: exchanges/kucoin/kucoin_test.go [hunk 1]
```diff
@@ -1156,11 +1156,19 @@ func TestGetFuturesContract(t *testing.T) {
...
-func TestGetFuturesRealTimeTicker(t *testing.T) {
-	t.Parallel()
-	_, err := ku.GetFuturesRealTimeTicker(context.Background(), "XBTUSDTM")
-	if err != nil {
-		t.Error("GetFuturesRealTimeTicker() error", err)
+func TestGetFuturesTicker(t *testing.T) {
+	t.Parallel()
+	tick, err := ku.GetFuturesTicker(context.Background(), "XBTUSDTM")
+	if assert.NoError(t, err, "GetFuturesTicker should not error") {
+		assert.Equal(t, "XBTUSDTM", tick.Symbol, "Symbol should be correct")
```
[the remark this answers is not in the document]
Comment (gbjk) [reply exchanges/kucoin/kucoin_test.go:1163]: Fixed and rebased down

**Comment by gbjk** [comment]
Just going to quickly fix the gci issue that gets caused by saving the test.

**Comment by gbjk** [comment]
Done. I've moved the json test into a fixture file to avoid the linter errors, and `jq -c`-ed it for consistency.
It's passing nicely and I've tested for false positives.

**Review by shazbert — APPROVED** [review]
changes ACK.

File: exchanges/kucoin/kucoin_test.go [hunk 2]
```diff
@@ -1156,11 +1156,20 @@ func TestGetFuturesContract(t *testing.T) {
...
 
-func TestGetFuturesRealTimeTicker(t *testing.T) {
+func TestGetFuturesTicker(t *testing.T) {
 	t.Parallel()
-	_, err := ku.GetFuturesRealTimeTicker(context.Background(), "XBTUSDTM")
-	if err != nil {
-		t.Error("GetFuturesRealTimeTicker() error", err)
+	tick, err := ku.GetFuturesTicker(context.Background(), "XBTUSDTM")
+	if assert.NoError(t, err, "GetFuturesTicker should not error") {
+		assert.Equal(t, "XBTUSDTM", tick.Symbol, "Symbol should be correct")
```
Comment (thrasher-) [inline exchanges/kucoin/kucoin_test.go:1163]: Looks like `assert.Positive(t, tick.Sequence, "Sequence should be positive")` was purged in 6488b08fb0c03c148eaaa3689dc12861c712bc94

File: exchanges/kucoin/kucoin_test.go [hunk 3]
```diff
@@ -1961,57 +1970,11 @@ func TestGetAuthenticatedServersInstances(t *testing.T) {
...
-	"Futures StartFundingFeeSettlementPushDataJSON":              `{ "topic": "/contract/announcement","subject": "funding.begin","data": {"symbol": "ETHUSDCM","fundingTime": 1551770400000,"fundingRate": -0.002966,"timestamp": 1551770400000}}`,
-	"Futures FundingRatePushDataJSON":                            `{ "topic": "/contract/instrument:ETHUSDCM","subject": "funding.rate","data": {"granularity": 60000,"fundingRate": -0.002966,"timestamp": 1551770400000}}`,
-	"Futures MarkIndexPricePushDataJSON":                         `{ "topic": "/contract/instrument:ETHUSDCM","subject": "mark.index.price","data": {"granularity": 1000,"indexPrice": 4000.23,"markPrice": 4010.52,"timestamp": 1551770400000}}`,
-	"Orderbook Market Level2":                                    `{ "type": "message", "topic": "/market/level2:BTC-USDT", "subject": "trade.l2update", "data": { "changes": { "asks": [ [ "18906", "0.00331", "14103845" ], [ "18907.3", "0.58751503", "14103844" ] ], "bids": [ [ "18891.9", "0.15688", "14103847" ] ] }, "sequenceEnd": 14103847, "sequenceStart": 14103844, "symbol": "BTC-USDT", "time": 1663747970273 } }`,
-}
+var websocketPushDatas = map[string]string{}
```
Comment (thrasher-) [inline exchanges/kucoin/kucoin_test.go:1973]: This is unused now

File: exchanges/kucoin/kucoin_test.go [hunk 4]
```diff
@@ -1961,57 +1970,11 @@ func TestGetAuthenticatedServersInstances(t *testing.T) {
...
 
 func TestPushData(t *testing.T) {
-	for key, val := range websocketPushDatas {
-		err := ku.wsHandleData([]byte(val))
-		if err != nil {
-			t.Errorf("%s: %v", key, err)
-		}
-	}
+	n := new(Kucoin)
+	sharedtestvalues.TestFixtureToDataHandler(t, ku, n, "testdata/wsHandleData.json", ku.wsHandleData)
```
Comment (thrasher-) [inline exchanges/kucoin/kucoin_test.go:1977]: This is throwing an error: https://github.com/thrasher-corp/gocryptotrader/actions/runs/7176138279/job/19540558917#step:7:3277

**Review by thrasher- — CHANGES_REQUESTED** [review]
Thanks for addressing the request! Found some additional issues

File: exchanges/kucoin/kucoin_test.go (hunk 2 above)
Comment (gbjk) [reply exchanges/kucoin/kucoin_test.go:1164]: Fixed. Thanks.

Suspect that was `rerere`

File: exchanges/kucoin/kucoin_test.go (hunk 4 above)
Comment (gbjk) [reply exchanges/kucoin/kucoin_test.go:1978]: That'd be the false positive test then 😂 
Damn, sorry, that was sloppy.
At least we know it works.

Fixed.

File: exchanges/kucoin/kucoin_test.go (hunk 3 above)
Comment (gbjk) [reply exchanges/kucoin/kucoin_test.go]: Fixed 54b61314

**Review by thrasher- — APPROVED** [review]
tACK, thanks!

**Review by gloriousCode — APPROVED** [review]
reTACK!

# 🌞

**Pull request merged**